core/services/relay/evm/mercury/queue.go

```diff
@@ -64,24 +64,28 @@ type Transmission struct {
 
 // maxlen controls how many items will be stored in the queue
 // 0 means unlimited - be careful, this can cause memory leaks
-func NewTransmitQueue(lggr logger.Logger, feedID string, maxlen int, transmissions []*Transmission, asyncDeleter asyncDeleter) *TransmitQueue {
-	pq := priorityQueue(transmissions)
-	heap.Init(&pq) // ensure the heap is ordered
+func NewTransmitQueue(lggr logger.Logger, serverURL, feedID string, maxlen int, asyncDeleter asyncDeleter) *TransmitQueue {
 	mu := new(sync.RWMutex)
 	return &TransmitQueue{
 		services.StateMachine{},
 		sync.Cond{L: mu},
 		lggr.Named("TransmitQueue"),
 		asyncDeleter,
 		mu,
-		&pq,
+		nil, // pq needs to be initialized by calling tq.Init before use
 		maxlen,
 		false,
 		nil,
-		transmitQueueLoad.WithLabelValues(feedID, fmt.Sprintf("%d", maxlen)),
+		transmitQueueLoad.WithLabelValues(feedID, serverURL, fmt.Sprintf("%d", maxlen)),
 	}
 }
 
+func (tq *TransmitQueue) Init(transmissions []*Transmission) {
```

Comment:
should this be wrapped in a do once?

Reply:
Hmm, no I don't think so. It's up to the caller. Presumably they could Init twice if they wanted.

Reply:
i only see this called in tests. Is it missing from somewhere else? should there be a guard somewhere that fails cleanly if init hasn't been called? right now i'm concerned there is a nil pointer panic lurking when someone forgets to init